File: p010/010.py
import math
s = 0


def checkPrime(num):
    root = math.sqrt(num)
    for i in range(2, root):  # todo: Fix this
        if(num % i == 0):
            return False
    return True


for i in range(2, 2000000):
    if checkPrime(i == True):
        s += i

print("Sum is", str(s))

File: p010/010.js
/**
 * @author Alexi Jamal
 *
 * @description
 * The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
 *
 * Find the sum of all the primes below two million.
 */

let sum = 0;
for (let i = 2; i < 2000000; i++) {
  if (checkPrime(i) == true) {
    sum += i;
  }
}

function checkPrime(num) {
  //! REMOVED BECAUSE IT IS SLOW
  // for (let i = 2; i < num; i++) {
  //   if (num % i === 0) return false;
  // }
  // return num > 1;
  let root = Math.sqrt(num);
  for (let i = 2; i <= root; i++) {
    if (num % i == 0) return false;
  }
  return true;
}

console.log(sum);
